warehouse/transformer/transformer.go

```diff
@@ -0,0 +1,205 @@
+package transformer
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+	"github.com/rudderlabs/rudder-go-kit/stats"
+
+	ptrans "github.com/rudderlabs/rudder-server/processor/transformer"
+	"github.com/rudderlabs/rudder-server/warehouse/transformer/internal/utils"
+)
+
+type (
+	transformer struct {
+		now func() time.Time
+
+		conf         *config.Config
+		logger       logger.Logger
+		statsFactory stats.Stats
+
+		config struct {
+			enableIDResolution           config.ValueLoader[bool]
+			enableArraySupport           config.ValueLoader[bool]
+			populateSrcDestInfoInContext config.ValueLoader[bool]
+			maxColumnsInEvent            config.ValueLoader[int]
+		}
+	}
+
+	processingInfo struct {
+		event         ptrans.TransformerEvent
+		itrOpts       integrationsOptions
+		dstOpts       destConfigOptions
+		jsonPathsInfo jsonPathInfo
+	}
+)
+
+func New(conf *config.Config, logger logger.Logger, statsFactory stats.Stats) ptrans.DestinationTransformer {
+	t := &transformer{
+		conf:         conf,
+		logger:       logger,
+		statsFactory: statsFactory,
+		now:          time.Now,
+	}
+
+	t.config.enableIDResolution = conf.GetReloadableBoolVar(false, "Warehouse.enableIDResolution")
+	t.config.enableArraySupport = conf.GetReloadableBoolVar(false, "Warehouse.clickhouse.enableArraySupport")
+	t.config.populateSrcDestInfoInContext = conf.GetReloadableBoolVar(true, "Warehouse.populateSrcDestInfoInContext")
+	t.config.maxColumnsInEvent = conf.GetReloadableIntVar(200, 1, "Warehouse.maxColumnsInEvent")
+	return t
+}
+
+func (t *transformer) Transform(_ context.Context, clientEvents []ptrans.TransformerEvent, _ int) (res ptrans.Response) {
+	if len(clientEvents) == 0 {
+		return
+	}
+
+	startTime := t.now()
+	metadata := clientEvents[0].Metadata
+
+	defer func() {
+		tags := stats.Tags{
```

Comment:
should we add a tag `implementation` to distinguish between `go` or `js-client`

Reply:
We are already prefixing `warehouse_` as part of the stat name.